rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/ConnectorImpl.java

```diff
@@ -47,9 +47,11 @@ public RxBleConnection call() throws Exception {
                 final Observable<RxBleConnection> disconnectedErrorObservable = connectionComponent.gattCallback().observeDisconnect();
                 final DisconnectAction disconnect = connectionComponent.disconnectAction();
 
-                return newConnectionObservable
-                        .delaySubscription(connectedObservable)
-                        .mergeWith(disconnectedErrorObservable)
+                return Observable.merge(
+                        newConnectionObservable.delaySubscription(connectedObservable),
+                        disconnectedErrorObservable,
+                        connectionComponent.mtuWatcherCompletable().<RxBleConnection>toObservable()
```

Comment:
This is acceptable but I'm thinking about some better way. This introduces API of a `Completable` that doesn't do anything but stays subscribed. Is there any good way to watch for connect/disconnect events and notify `MtuWatcher` to subscribe/unsubscribe from the MTU changes? This would be more descriptive that merging with a magic` Completable`.

Reply:
I just try to avoid manual state management

Reply:
Why is manual state management bad? I think it might be more descriptive.

If you want to stick with this solution I would name the method some other way. It should clearly say that it is responsible for the watcher state.